pkg/client/e2e.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/results"
+	"github.com/heptio/sonobuoy/pkg/results/e2e"
+	"github.com/onsi/ginkgo/reporters"
+	"github.com/pkg/errors"
+)
+
+// GetTests extracts the junit results from a sonobuoy archive and returns the requested tests.
+func (c *SonobuoyClient) GetTests(reader io.Reader, show string) ([]reporters.JUnitTestCase, error) {
+	read := results.NewReaderWithVersion(reader, "irrelevant")
+	junitResults := reporters.JUnitTestSuite{}
+	err := read.WalkFiles(func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		// TODO(chuckha) consider reusing this function for any generic e2e-esque plugin results.
+		return results.ExtractFileIntoStruct(results.PluginsDir+e2e.ResultsSubdirectory+e2e.JUnitResultsFile, path, info, &junitResults)
```

Comment:
could you use paths.Join() here?

Reply:
I'll make a TODO because this will require some other refactoring and since sonobuoy is in control of writing the tarball, it doesn't seem too likely that it will bite us (famous last words).